spec/ApiClient/SearchClientSpec.php

```diff
@@ -25,16 +25,31 @@ public function let(HttpClient $httpClient)
     public function it_queries()
     {
         $request = new Request('GET',
-            'search?for=foo&page=1&per-page=20&sort=date&order=desc&subject[]=cell-biology&type[]=research-article&use-date=published&start-date=2017-01-02&end-date=2017-02-03',
+            'search?for=foo/%23%20bar&page=1&per-page=20&sort=date&order=desc&subject[]=cell-biology&type[]=research-article&use-date=published&start-date=2017-01-02&end-date=2017-02-03',
             ['X-Foo' => 'bar', 'Accept' => 'application/vnd.elife.search+json; version=2', 'User-Agent' => 'eLifeApiClient/'.Version::get()]);
         $response = new FulfilledPromise(new ArrayResult(new MediaType('application/vnd.elife.search+json',
             2), ['foo' => ['bar', 'baz']]));
 
         $this->httpClient->send($request)->willReturn($response);
 
-        $this->query((['Accept' => 'application/vnd.elife.search+json; version=2']), 'foo', 1, 20, 'date', true,
+        $this->query((['Accept' => 'application/vnd.elife.search+json; version=2']), 'foo/# bar', 1, 20, 'date', true,
             ['cell-biology'], ['research-article'], 'published', new DateTimeImmutable('2017-01-02'), new DateTimeImmutable('2017-02-03'))
             ->shouldBeLike($response)
         ;
     }
+
+    public function it_always_queries_with_for()
+    {
+        $request = new Request('GET',
+            'search?page=1&per-page=20&sort=relevance&order=desc&use-date=default&for=',
+            ['X-Foo' => 'bar', 'Accept' => 'application/vnd.elife.search+json; version=2', 'User-Agent' => 'eLifeApiClient/'.Version::get()]);
+        $response = new FulfilledPromise(new ArrayResult(new MediaType('application/vnd.elife.search+json',
+            2), ['foo' => ['bar', 'baz']]));
+
+        $this->httpClient->send($request)->willReturn($response);
+
+        $this->query((['Accept' => 'application/vnd.elife.search+json; version=2']))
+            ->shouldBeLike($response)
+        ;
+    }
```

Comment:
is there a test here that demonstrates the new functionality? Most of the code seems related to the required `for` parameter rather than percentage-encoding